lib/matplotlib/dates.py

```diff
@@ -430,6 +430,9 @@ def date2num(d):
     For details see the module docstring.
     """
 
+    if hasattr(d, "to_pydatetime"):
+        d = d.to_pydatetime()
```

Comment:
I think we are unnecessarily losing efficiency here; to avoid that, use the same 2-part conditional here as you use in get_converter.  That way, if the input is basically datetime64 it can be handled by your nice, efficient, fully vectorized converter instead of making a round-trip through the clunky datetime.datetime machinery.

Reply:
Can pandas go to datetime64?  Ity my understanding that the underlying data type is datetime64[ns], but I don't know what attribute to test for to get that to come out...

Reply:
Oh, duh, I see...